services/src/test/scala/cromwell/services/instrumentation/impl/statsd/StatsDInstrumentationServiceActorSpec.scala

```diff
@@ -70,11 +80,37 @@ class StatsDInstrumentationServiceActorSpec extends TestKitSuite with FlatSpecLi
         "prefix_value.cromwell.test_prefix.test.metric.bucket.timing.mean:5000.00|g"
       ),
       Set("prefix_value.cromwell.test_prefix.test.metric.bucket.timing.m1_rate")
+    ),
+    StatsDTestBit("increment counters with cromwell id prefix", CromwellIncrement(testBucket),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.samples:1|g"),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.m1_rate"),
+      prependCromwellInstancePrefix = true
+    ),
+    StatsDTestBit("add count with cromwell id prefix", CromwellCount(testBucket, 80),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.samples:81|g"),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.m1_rate"),
+      prependCromwellInstancePrefix = true
+    ),
+    StatsDTestBit("set gauges with cromwell id prefix", CromwellGauge(testGaugeBucket, 89),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.gauge.metric.bucket:89|g"),
+      prependCromwellInstancePrefix = true
+    ),
+    StatsDTestBit("set timings with cromwell id prefix", CromwellTiming(testBucket.expand("timing"), 5.seconds),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.timing.stddev:0.00|g",
+        "prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.timing.samples:1|g",
+        "prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.timing.p95:5000.00|g",
+        "prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.timing.mean:5000.00|g"
+      ),
+      Set("prefix_value.cromwell.cromwell-101.test_prefix.test.metric.bucket.timing.m1_rate"),
+      prependCromwellInstancePrefix = true
     )
   ) foreach {
-    case StatsDTestBit(description, metric, expectedExactPackets, expectedFuzzyPackets) =>
+    case StatsDTestBit(description, metric, expectedExactPackets, expectedFuzzyPackets, prependCromwellInstancePrefix) =>
       it should description in {
-        val instrumentationActor = TestActorRef(new StatsDInstrumentationServiceActor(config, ConfigFactory.load(), registryProbe))
+
+        val globalConfig = if (prependCromwellInstancePrefix) cromwellInstanceConfig else ConfigFactory.load()
```

Comment:
This is so weird... and existing code! I haven't looked to closely, but the fact that a config with _only_ the `system.cromwell_id` key is passing tests implies that this `ConfigFactory.load()` could be simplified to `ConfigFactory.empty()`.

Anyway, if you want to add your `system.cromwell_id` to the `ConfigFactory.load()` instead of replacing the whole thing, use something like:

```scala
ConfigFactory.load().withValue("system.cromwell_id", ConfigValueFactory.fromAnyRef("cromwell-101"))
```

So, either both of values in the `if` should be full-ish configs, or both should be empty-ish.

Reply:
I am not 100% sure, but the `StatsDInstrumentationServiceActor` only requires `globalConfig` to extract the `system.cromwell_id` now. Before I don't think it was used at all. So I guess we can replace `ConfigFactory.load()` with `ConfigFactory.empty()` no? Or should we still load the default config?

Reply:
If tests pass, I would just use `empty()`.

Otherwise, here's another alternative to the `withValue()` above:

```scala
val cromIdConfig = if (prependCromwellInstancePrefix) cromwellInstanceConfig else ConfigFactory.empty()
val globalConfig = cromIdConfig.withFallBack(ConfigFactory.load())
```

Reply:
Yes, the tests do pass with `ConfigFactory.empty()`. So I will go ahead with it 👍